src/main/java/se/ayoy/maven/plugins/licenseverifier/LicenseAbstractMojo.java

```diff
@@ -96,31 +98,52 @@
                 continue;
             }
 
-            AyoyArtifact licenseInfo = new AyoyArtifact(artifact);
+            toReturn.add(toAyoyArtifact(artifact, buildingRequest));
 
-            getLog().debug("Getting license for " + artifact.toString());
-            try {
-                buildingRequest.setProject(null);
+            Set<Artifact> transitiveArtifacts = resolveTransitiveArtifact(artifact);
 
-                MavenProject mavenProject = projectBuilder.build(artifact, buildingRequest).getProject();
-                if (mavenProject == null) {
-                    throw new NullPointerException("MavenProject retrieved from ProjectBuilder.build is null");
-                }
+            StringBuilder transitiveArtifactsList = new StringBuilder();
+            for (Artifact transitiveArtifact : transitiveArtifacts) {
+                toReturn.add(toAyoyArtifact(transitiveArtifact, buildingRequest));
+                transitiveArtifactsList.append(lineSeparator())
+                        .append(transitiveArtifact.toString());
+            }
+            if (getLog().isDebugEnabled() && !transitiveArtifacts.isEmpty()) {
+                getLog().debug("Verifying "
+                        + transitiveArtifacts.size()
+                        + " transitive artifacts for "
+                        + artifact.toString()
+                        + ":"
+                        + transitiveArtifactsList
+                );
+            }
+        }
 
-                List<License> licenses = mavenProject.getLicenses();
-                if (licenses == null) {
-                    throw new NullPointerException("Licenses is null, from mavenProject from " + artifact);
-                }
+        return toReturn;
+    }
 
-                licenseInfo.addLicenses(licenses);
+    private AyoyArtifact toAyoyArtifact(Artifact artifact, ProjectBuildingRequest buildingRequest) {
+        AyoyArtifact licenseInfo = new AyoyArtifact(artifact);
 
-                toReturn.add(licenseInfo);
-            } catch (ProjectBuildingException e) {
-                getLog().error(e.getMessage());
+        getLog().debug("Getting license for " + artifact.toString());
+        try {
+            buildingRequest.setProject(null);
+
+            MavenProject mavenProject = projectBuilder.build(artifact, buildingRequest).getProject();
+            if (mavenProject == null) {
+                throw new NullPointerException("MavenProject retrieved from ProjectBuilder.build is null");
             }
-        }
 
-        return toReturn;
+            List<License> licenses = mavenProject.getLicenses();
+            if (licenses == null) {
+                throw new NullPointerException("Licenses is null, from mavenProject from " + artifact);
```

Comment:
here it could be 
`Objects.requireNonNull(licenses, "Licenses is null, from mavenProject from " + artifact)`
from Java 1.7.

Reply:
Using Objects.requireNonNull looks good.